### PR TITLE
Simplified flowers' registration

### PR DESCRIPTION
**Pull request by LeMagnesium**
- Created a new local function to register flowers

This commit is the final form of the pull-request #492 which was... more a draft than anything else.

EDIT: I'm trying to sign the commit as @Ombridride , I just have to ask him his mail before the PR can be merged.
EDIT2: Done.

### CODE REVIEW

**Comment by HybridDog** [comment]
You could reduce the code by using a table.

**Comment by LeMagnesium** [comment]
Yes, I didn't see it go when I copied @Ombridride 's pull request's code.
There is the commit : https://github.com/LeMagnesium/minetest_game/commit/6140b718cf7eb9562159668edb423e2f1c4bd177
 - All flowers' datas are in a table
 - All common groups are added in the `f_groups` table in the function
 - The missing namespace is back

**Comment by HybridDog** [comment]
and the texture_prefix is "flowers_"..name

**Comment by LeMagnesium** [comment]
Very good advise. Then see https://github.com/LeMagnesium/minetest_game/commit/fda17fa7a509a4c8caccf73e79c098634db2f7b9 .

**Comment by PilzAdam** [comment]
:+1:

File: mods/flowers/init.lua [hunk 1]
```diff
@@ -15,113 +15,50 @@ minetest.register_alias("flowers:flower_rose", "flowers:rose")
...
-		type = "fixed",
-		fixed = { -0.5, -0.5, -0.5, 0.5, -0.2, 0.5 },
-	},
-})
+for _,item in pairs(flowers.datas) do
+	name = item[1]
+	description = item[2]
+	groups = item[3]
+	add_simple_flower(name, description, groups)
+end
```
Comment (HybridDog) [inline mods/flowers/init.lua]: Does ```add_simple_flower(unpack(item))``` work too?

File: mods/flowers/init.lua [hunk 2]
```diff
@@ -15,113 +15,50 @@ minetest.register_alias("flowers:flower_rose", "flowers:rose")
...
-	selection_box = {
-		type = "fixed",
-		fixed = { -0.5, -0.5, -0.5, 0.5, -0.2, 0.5 },
-	},
-})
+for _,item in pairs(flowers.datas) do
+	name = item[1]
+	description = item[2]
+	groups = item[3]
+	add_simple_flower(name, description, groups)
```
Comment (HybridDog) [inline mods/flowers/init.lua]: Does ```add_simple_flower(unpack(item))``` work too?

Comment (LeMagnesium) [inline mods/flowers/init.lua]: It does, and also thanks, you taught me a new (useful) method.
https://github.com/LeMagnesium/minetest_game/commit/5de52e242a9b9e096d879ae926d8b647773e8f5c

**Comment by Novatux** [comment]
+1

**Comment by Ombridride** [comment]
I think the work on this code to improve the "flowers mod" of MT_game is over, maybe you could merge it ?

**Comment by LeMagnesium** [comment]
@Ombridride : The devs must review the code, two of them have agreed on it so far, but one did before I modify something, so only one out of the two needed agreements is obtained for this PR.

**Comment by LeMagnesium** [comment]
I fixed a bug reported by @mgl512 , about the new selection boxes of white dandelions and violas.
Should I squash my commits?

**Comment by Novatux** [comment]
https://github.com/minetest/minetest_game/commit/01ad09036b73875639f70ed9caad1dd830183a98
https://github.com/minetest/minetest_game/commit/638add603f56d8797cae8563839e9eff65a8c8e8

**Pull request closed without merging**